go-controller/pkg/ovn/controller/services/load_balancer.go

```diff
@@ -0,0 +1,390 @@
+package services
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	globalconfig "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	ovnlb "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/loadbalancer"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	v1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1beta1"
+	"k8s.io/klog/v2"
+)
+
+// lbConfig is the abstract desired load balancer configuration.
+// vips and endpoints are mixed families.
+type lbConfig struct {
+	vips     []string // just ip or the special value "node" for the node's physical IPs (i.e. NodePort)
+	protocol v1.Protocol
+	inport   int32
```

Comment:
the input port. I felt that the "target" is ambiguous - is it the target of the LB, or the target of the backends?

If you mentally model the LB as a multiplexer, then input -> output makes sense. But I see what you mean, that's not necessarily clear.

If you have better verbiage, then I'm happy to update.

Reply:
ack, I see what you mean now when you say "input" into LB.. I think we can stick to it since its internal anyways, if you happen to update the PR, a comment like you have for the vips will do.

Reply:
Added a comment.